GunShipAlpha/src/Game.java

```diff
@@ -149,12 +146,18 @@ private Game.spawnEnemiesRandomLocation getRandomLocation(Random rand) {
         return getSpawnEnemiesRandomLocation(side, y, x, rand);
     }
 
-    private void spawnEnemyOfTypeAtLocation(boolean isTank, int x, int y) {
-        if (isTank) {
-            spawnTankAt(x, y);
-        } else {
+    private void spawnEnemyOfTypeAtLocation(int enemyType, int x, int y) {
```

Comment:
here's an idea that might blow your mind: instead of passing the enemyType, you could pass in the _class_. You'll need some help from chatgpt on how to use the class to implement spawnEnemyAt but it is doable, and will remove all this duplication.

neat!